### ISSUE TITLE
Cannot use rbenv / ruby_build on centos 6.3 (in Vagrant and in EC2)

### ISSUE THREAD

**Issue by scottbessler** [sous-chefs/ruby_rbenv#39]
I have a recipe with the following line in a recipe:

```
rbenv_global "1.9.3-p327"
```

And I have tried including all permutations of the following 3 recipes in my run list to make it work (on both Vagrant with a centos 6.3 box and in EC2 with a centos 6.3 ami):

```
rbenv::system
rbenv::system_install
rbenv::vagrant
```

I receive the following error:

```
[2013-03-13T16:43:48+00:00] FATAL: NameError: Cannot find a resource for rbenv on centos version 6.3
```

**Comment by johnbellone** [comment]
Where is this actually coming from? Are you sure that you have it in a role or run_list in the actual Vagrantfile?

You have this in a cookbooks path which is exposed there as well, right?

**Comment by scottbessler** [comment]
That is a message from Chef. I've tried 2 methods:
1. vagrant+chef-solo, where rbenv and ruby_build cookbooks are in the cookbook path and the items I listed in the issue are in the runlist
2. an ec2 instance talking to a chef server, where rbenv and ruby_build are in the run list

Its been a while now and we worked around using our own chef code to set up rbenv/ruby_build but I can take a look again if you have more questions at some point.

**Comment by johnbellone** [comment]
You shouldn't have to include `rbenv::system_install` as per the documentation `rbenv::system` includes this recipe. But that shouldn't be the error you're seeing. I've only seen this when my Vagrantfile didn't have the recipe and it was due to me not performing a `git submodule update --init` on initial checkout (obviously that's a result of how we packaged our definitions). 

Likely perhaps something that's loading up earlier in the run_list is requiring rbenv? I separated this dependency by creating a role specifically for Ruby applications and rather include that in the run_list. I would take a look at the preceding roles and recipes to try to isolate what may be requiring rbenv. Sometimes cookbooks have several things in a literal array statement that will add dependencies in the metadata.rb file.
